### ISSUE TITLE
Runtime error when isArray.extracting.contains with multiple fields (tuples)

### ISSUE THREAD

**Issue by mitasov-ra** [lukas-krecan/JsonUnit#493]
**Describe the bug**
1. AssertJ v2.32.0
2. Create a small reproducible example and paste it the the bug report.
```java
assertThatJson("""
    [
      {"id": 1, "name":"Aaron"},
      {"id": 2, "name":"Paul"}
    ]
    """)
    .isArray()
    .extracting("id", "name")
    .contains(
        tuple(1, "Aaron"),
        tuple(1, "Aaron")
    );
```
results in:

```
Unsupported type class org.assertj.core.groups.Tuple
java.lang.IllegalArgumentException: Unsupported type class org.assertj.core.groups.Tuple
	at net.javacrumbs.jsonunit.core.internal.GenericNodeBuilder.newNode(GenericNodeBuilder.java:63)
	at net.javacrumbs.jsonunit.core.internal.GenericNodeBuilder.wrapDeserializedObject(GenericNodeBuilder.java:34)
	at net.javacrumbs.jsonunit.core.internal.JsonUtils.wrapDeserializedObject(JsonUtils.java:181)
	at net.javacrumbs.jsonunit.assertj.JsonComparator.compare(JsonComparator.java:51)
	at org.assertj.core.internal.ComparatorBasedComparisonStrategy.iterableContains(ComparatorBasedComparisonStrategy.java:91)
	at org.assertj.core.internal.Iterables.iterableContains(Iterables.java:359)
	at org.assertj.core.internal.Iterables.lambda$assertIterableContainsGivenValues$0(Iterables.java:351)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:178)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:992)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
	at org.assertj.core.internal.Iterables.assertIterableContainsGivenValues(Iterables.java:352)
	at org.assertj.core.internal.Iterables.assertContains(Iterables.java:346)
	at org.assertj.core.api.AbstractIterableAssert.containsForProxy(AbstractIterableAssert.java:366)
	at org.assertj.core.api.AbstractIterableAssert.contains(AbstractIterableAssert.java:359)
```

3. Describe expected behavior

It just works.

**Comment by lukas-krecan** [comment]
Thanks, I will look at it

**Comment by lukas-krecan** [comment]
Fixed released as 2.33.0, thanks for reporting
